src/components/Table/GridTable.tsx

```diff
@@ -1214,39 +1203,29 @@ function useToggleIds(rows: GridDataRow<Kinded>[], persistCollapse: string | und
   const [tick, setTick] = useState<string>("");
 
   // Create the stable `toggleId`, i.e. we are purposefully passing an (almost) empty dep list
-  const toggleId = useCallback(
-    (id: string) => {
+  const toggleAll = useCallback(
+    (_id: string) => {
       // We have different behavior when going from expand/collapse all.
-      if (id === "header") {
-        const isAllCollapsed = collapsedIds[0] === "header";
-        collapsedIds.splice(0, collapsedIds.length);
-        if (isAllCollapsed) {
-          // Expand all means keep `collapsedIds` empty
-        } else {
-          // Otherwise push `header` on the list as a hint that we're in the collapsed-all state
-          collapsedIds.push("header");
-          // Find all non-leaf rows so that toggling "all collapsed" -> "all not collapsed" opens
-          // the parent rows of any level.
-          const parentIds = new Set<string>();
-          const todo = [...rows];
-          while (todo.length > 0) {
-            const r = todo.pop()!;
-            if (r.children) {
-              parentIds.add(r.id);
-              todo.push(...r.children);
-            }
-          }
-          // And then mark all parent rows as collapsed.
-          collapsedIds.push(...parentIds);
-        }
+      const isAllCollapsed = collapsedIds[0] === "header";
+      collapsedIds.splice(0, collapsedIds.length);
```

Comment:
I do not know the initial reasoning for this 🤔

Reply:
@bdow `getCollapsedRows` afaiu would return a new list instance every time, but we want the identity of `collapseIds` to never change, b/c the `useCallback`-d / `useMemo`-d `toggleId` functions capture it.

I.e. if we didn't capture it / didn't mutate a shared instance, then we would have a new `collapseIds` passed to `GridRow` every time, so collapsing a single row would invalidate all of the rows in the table, instead of just invalidating the currently toggled row.

Reply:
Gotcha, so that is also why we do not need to add `collapsedIds` to our hook dependency arrays as well, correct?

Reply:
Yep!